### ISSUE TITLE
Casting object columns with nulls (NaN) to strings causes "nan" string values rather than nulls

### ISSUE THREAD

**Issue by isrsal** [awslabs/aws-data-wrangler#362]
**Describe the bug**
A pandas dataframe with an `object` dtype column (default) for which a `string` type is passed to `s3.to_parquet` dtype parameter, null values (represented as NaN in pandas) are converted to "nan" strings instead of nulls

**To Reproduce**
python 3.8
wrangler 1.8.1
```
import pandas as pd
import awswrangler as wr
data = [{'A':'valueA'}, {'A': 'valueA', 'B':'valueB'}]
df = pd.DataFrame(data)
wr.s3.to_parquet(df, 's3://mybucket/nan_test/', dataset=True, database='mydb', table='nan_test', dtype={'A':'string', 'B': 'string'})
```

```
select b from mydb
```

**Comment by igorborgest** [comment]
Released on version **[1.9.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.9.0)**

**Comment by bruno-uy** [comment]
This ticket solved one issue that I had some dev. Thanks 😄
